src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
My concern is that this is a throw-away Response object: it will only be used by receive middleware, but not be available to the listener callback. This means, for example, that any attempts to inject custom response interception (response middleware) via receive middleware are going to fail.

I don't have a solution to `res.reply`, but I think that having a unique interface is okay. Each middleware stack serves a completely different purpose and will have different information available to it:
* Receive middleware: message
* Listener middleware: response, listener
* Response middleware: envelope